extensions/amp-analytics/0.1/events.js

```diff
@@ -168,6 +188,24 @@ export class CustomEventTracker extends EventTracker {
    * @param {!AnalyticsEvent} event
    */
   trigger(event) {
+    const observers = this.observers_[event.type];
+    if (startsWith(event.type, 'sandbox')) {
+      // special sandboxed event type
+      if (!this.sandboxBuffer_) {
+        return;
+      }
+
+      if (this.sandboxBuffer_[event.type]) {
```

Comment:
I tried but not that much...  The only part that can shared is push to buffer if buffer exists. But checking if buffer exists, or create buffer seems to have different logic. sharing one line `buffer.push(event)` doesn't sound so necessary then.